x-pack/plugins/synthetics/server/synthetics_service/formatters/private_formatters/processors_formatter.ts

```diff
@@ -6,30 +6,29 @@
  */
 
 import { ProcessorFields } from './format_synthetics_policy';
-import { MonitorFields } from '../../../../common/runtime_types';
-
-type Fields = Record<string, string | boolean>;
+import { HeartbeatFields, MonitorFields } from '../../../../common/runtime_types';
 
 interface FieldProcessor {
   add_fields: {
     target: string;
-    fields: Fields;
+    fields: HeartbeatFields;
   };
 }
 
 export const processorsFormatter = (config: Partial<MonitorFields & ProcessorFields>) => {
-  const fields: Fields = {
+  const fields: HeartbeatFields = {
     'monitor.fleet_managed': true,
+    config_id: config.config_id!,
+    meta: {
+      space_id: config.spaceId!,
```

Comment:
Shouldn't you handle the case where spaceId and config_id are undefined instead of forcing its types with `!`